File: map/123/src/main/java/Repos/Repository.java
package Repos;

import Models.Validators.ValidationException;

import java.util.Optional;


/**
 * @param <E> tipul entitatii care osa contina
 * @param <ID> tipul idiului entitatii
 * Interfata care ofera operatii CRUD
 */
public interface Repository<E, ID> {

    long size();
    Optional<E> save(E entity) throws ValidationException;
    Optional<E> delete(ID id);
    Optional<E> update(E entity) throws ValidationException;
    Optional<E> find(ID id);
    Iterable<E> getAll();
}

File: map/Lab7Map/Lab7Map/Repository/Repository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab7Map.Repository
{
    public interface IRepository<ID,E>
    {
        long Size();
        E Save(E entity);
        E Update(E entity);
        E Delete(ID id);
        E Find(ID id);
        List<E> FindAll();
    }
    
}
